src/components/FloatingLabel/FloatingLabel.tsx

```diff
@@ -0,0 +1,448 @@
+import type { ComponentProps } from 'react';
+import { forwardRef, useId } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { mergeDeep } from '~/src/helpers/merge-deep';
+import type { DeepPartial, FlowbiteSizes } from '../../';
+import { useTheme } from '../../';
+
+export interface FlowbiteFloatingLabelTheme {
+  filled_error?: string;
+  outlined_error?: string;
+  standard_error?: string;
+  filled_success?: string;
+  outlined_success?: string;
+  standard_success?: string;
+  filled_error_sm?: string;
+  outlined_error_sm?: string;
+  standard_error_sm?: string;
+  filled_success_sm?: string;
+  outlined_success_sm?: string;
+  standard_success_sm?: string;
+  filled_success_label?: string;
+  outlined_success_label?: string;
+  standard_success_label?: string;
+  filled_success_label_sm?: string;
+  outlined_success_label_sm?: string;
+  standard_success_label_sm?: string;
+  filled_error_label?: string;
+  outlined_error_label?: string;
+  standard_error_label?: string;
+  filled_error_label_sm?: string;
+  outlined_error_label_sm?: string;
+  standard_error_label_sm?: string;
+  success_helptext?: string;
+  error_helptext?: string;
+  default_filled?: string;
+  default_filled_label?: string;
+  default_outlined?: string;
+  default_outlined_label?: string;
+  default_standard?: string;
+  default_standard_label?: string;
+  default_helperText?: string;
+  default_filled_sm?: string;
+  default_filled_label_sm?: string;
+  default_outlined_sm?: string;
+  default_outlined_label_sm?: string;
+  default_standard_sm?: string;
+  default_standard_label_sm?: string;
+}
+
+export interface FlowbiteFloatingLabelSizes extends Pick<FlowbiteSizes, 'sm' | 'md' | 'lg'> {
+  [key: string]: string;
+}
+
+export interface FloatingLabelProps extends Omit<ComponentProps<'input'>, 'ref' | 'color'> {
+  color?: string | null;
+  helperText?: string;
+  sizing?: keyof FlowbiteFloatingLabelSizes;
+  variant: string;
+  label: string;
+  disabled?: boolean;
+  theme?: DeepPartial<FlowbiteFloatingLabelTheme>;
+}
+
+export const FloatingLabel = forwardRef<HTMLInputElement, FloatingLabelProps>(
+  (
+    {
+      color = null,
+      helperText,
+      sizing = 'md',
+      variant,
+      label,
+      disabled = false,
+      className,
+      theme: customTheme = {},
+      ...props
+    },
+    ref,
+  ) => {
+    const theme = mergeDeep(useTheme().theme.floatingLabel, customTheme);
+    const randomId = useId();
+
+    if (sizing === 'md') {
```

Comment:
Hi @okelloMarvinSophany 
the idea of having the theme object is to avoid having these many conditional renderings.

I can see that you are duplicating the blocks of code with these `variant === 'xyz'`, etc. 

You can check the standard of the use of the theme object in the other components like TextInput and Button, which have different "styles" and colors, but do not repeat the block of code for each of the possible variations.

Reply:
Let me fix that